core/src/main/java/org/sonarsource/sonarlint/core/container/connected/WsHelperImpl.java

```diff
@@ -39,12 +43,12 @@ public ValidationResult validateConnection(ServerConfiguration serverConfig) {
     try {
       SonarLintWsClient client = new SonarLintWsClient(serverConfig);
       ServerVersionAndStatusChecker serverChecker = new ServerVersionAndStatusChecker(client);
-      ValidationResult result = serverChecker.validateStatusAndVersion();
-      if (result.success()) {
-        return new AuthenticationChecker(client).validateCredentials();
-      } else {
-        return result;
-      }
+      PluginVersionChecker pluginsChecker = new PluginVersionChecker(client);
+      ServerInfos serverInfo = serverChecker.checkVersionAndStatus();
+      pluginsChecker.checkPlugins(serverInfo.getVersion());
+      return new AuthenticationChecker(client).validateCredentials();
+    } catch (UnsupportedServerException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)